streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java

```diff
@@ -798,17 +798,43 @@ public void shouldReturnEmptyClusterMetadataIfItHasntBeenBuilt() throws Exceptio
     }
 
     @Test
-    public void shouldNotLoopInfinitelyOnMissingMetadataAndShouldNotCreateRelatedTasks() {
+    public void shouldCreateAllTasksAndInternalTopicIfAllInputTopicsAreKnown() {
+        runMetadataTest("topic1", true, "topic3", true);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks1() {
+        runMetadataTest("topic1", true, "unknownTopic", false);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks2() {
```

Comment:
Do we need to have 2 tests to cover the partly missing metadata scenario? Aren't they both ultimately testing the same thing? i.e., that we don't end up in an infinite loop if some metadata is missing?